acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+import datetime as dt
+import requests
+from bs4 import BeautifulSoup
+
+
+__taskname__ = "acszpt"
+__version__ = "1.0"
+__author__ = "Nathan Miles"
+__vdate__ = "22-Jan-2019"
+__all__ = ['acszpt']
+
+
+
+class Query(object):
+    """ Class used to interface with the ACS Zeropoints Calculator API.
+
+    Parameters
+    ----------
+    date : str
+        Input date in the following ISO format, YYYY-MM-DD
+    detector : str
+        One of the three channels on ACS: HRC, SBC, or WFC
+    filt : str, optional
+        One of valid filters for the chosen detector
+            * HRC:
+                 F220W, F250W, F330W,
+                 F344N, F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+
+            * WFC:
+                 F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+            * SBC:
+                F115LP, F122M, F125LP,
+                F140LP, F150LP, F165LP
+
+    Attributes
+    ----------
+    date : str
+        **Attribute** for storing the user specified date.
+    detector : str
+        **Attribute** for storing the user specified detector.
+    filt : str, None
+        **Attribute** for storing the user specified filter (if supplied).
+    zpt_table : astropy.table.Table
+        **Attribute** for storing the results returned by the ACS Zeropoint Calculator
+    """
+
+    def __init__(self, date, detector, filt=None):
+
+        # Set the public attributes
+        self.date = date
+        self.detector = detector.upper()
+        self.filt = filt
+
+        self.valid_filters = {
+            'WFC': ['F435W', 'F475W', 'F502N', 'F550M',
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'HRC': ['F220W', 'F250W', 'F330W', 'F344N',
+                    'F435W', 'F475W', 'F502N', 'F550M',
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'SBC': ['F115LP', 'F122M', 'F125LP',
+                   'F140LP', 'F150LP', 'F165LP']
+        }
+        self.zpt_table = None
+
+        # Set the private attributes
+        if filt is None:
+            self._url = 'https://acszeropoints.stsci.edu/results_all/?date={}&' \
+                        'detector={}'.format(self.date, self.detector)
+        else:
+            self.filt = filt.upper()
+
+            self._url = 'https://acszeropoints.stsci.edu/results_single/' \
+                        '?date1={0}&detector={1}&{1}_filter={2}'. \
+                format(self.date, self.detector, self.filt)
+
+        self._zpts = OrderedDict()
+        self._response = None
+        self._failed = False
+
+    def _check_inputs(self):
+        """ Check the input to ensure they are valid
+
+        Returns
```

Comment:
This looks like an incomplete docstring. It seems to return `bool` but that is not documented.